src/ring_theory/adjoin_root.lean

```diff
@@ -168,6 +168,21 @@ by rw [is_root, eval_map, eval₂_root]
 lemma is_algebraic_root (hf : f ≠ 0) : is_algebraic R (root f) :=
 ⟨f, hf, eval₂_root f⟩
 
+lemma mk_injective_of_degree_ne_zero [is_domain R] (hf : f.degree ≠ 0) :
+  function.injective (adjoin_root.of f) :=
+begin
+  rw ring_hom.injective_iff_ker_eq_bot,
+  refine le_antisymm (λ p hp, _) bot_le,
+  rw [ring_hom.mem_ker, adjoin_root.of, ring_hom.comp_apply, adjoin_root.mk_eq_zero] at hp,
+  by_cases h : f = 0,
+  { refine C_eq_zero.mp (eq_zero_of_zero_dvd (by rwa h at hp)) },
+  { by_contra h_contra,
+    refine hf _,
+    rw ← degree_C h_contra,
+    apply le_antisymm (degree_le_of_dvd hp (by rwa [ne.def, C_eq_zero, ← ideal.mem_bot])) _,
+    rwa [degree_C h_contra, zero_le_degree_iff] },
+end
```

Comment:
Slightly golfed: we don't need to go through `ker = ⊥`, we can directly use `f x = 0 → x = 0`:
```suggestion
begin
  rw injective_iff_map_eq_zero,
  intros p hp,
  rw [adjoin_root.of, ring_hom.comp_apply, adjoin_root.mk_eq_zero] at hp,
  by_cases h : f = 0,
  { exact C_eq_zero.mp (eq_zero_of_zero_dvd (by rwa h at hp)) },
  { contrapose! hf with h_contra,
    rw ← degree_C h_contra,
    apply le_antisymm (degree_le_of_dvd hp (by rwa [ne.def, C_eq_zero])) _,
    rwa [degree_C h_contra, zero_le_degree_iff] },
end
```

Reply:
I'd like to extract a lemma saying `f | C p → degree f ≤ 0`. But I'd also be happy to put that in a follow-up.

Reply:
Sure I'll extract that in a follow-up!